src/js/Controllers/TTSController.js

```diff
@@ -76,16 +83,48 @@ class TTSController {
                 } else if (this.filePlaylist[0].type === "TEXT"){
                     this.speak();
                 }    
+            } else {
+                this.speakEnded();
             }
         }
 
+        this.currentlyPlaying = "TEXT";
         speechPlayer.text = text;
         speechPlayer.volume = 1;
         speechPlayer.rate = 1;
         speechPlayer.pitch = 0;
         window.speechSynthesis.speak(speechPlayer)
 
     }
+
+    stopSpeak(stopSpeakingID) {
+        const speakID = this.speakID;
+        this.speakID = null;
+        if (this.currentlyPlaying === "FILE") {
+            this.audioPlayer.onended = null;
+            this.audioPlayer.pause();
+            this.audioPlayer.src = "";
+        } else if (this.currentlyPlaying === "TEXT") {
+            window.speechSynthesis.pause();
+            window.speechSynthesis.cancel();
+        }
+        this.filePlaylist = [];
+        this.currentlyPlaying = null;
+        this.listener.send(RpcFactory.TTSStopSpeakingSuccess(stopSpeakingID));
```

Comment:
Based on the documentation I wanted to have StopSpeaking respond before Speak and Stopped messages https://smartdevicelink.com/en/docs/hmi/master/tts/stopspeaking/

Reply:
Ah, seeing that this actually would cause issues on Core's end. Still feels a bit weird to be sending all of these messages in this function, but I'm not seeing a cleaner way to do it for now.